firehose_to_redshift/main.tf

```diff
@@ -0,0 +1,303 @@
+resource "aws_cloudwatch_log_group" "log_group" {
+    name = "/aws/kinesisfirehose/${var.env_name}-${var.stream_name}"
+    retention_in_days = "${var.log_retention_in_days}"
+
+    tags {
+        environment = "${var.env_name}"
+    }
+}
+
+resource "aws_cloudwatch_log_stream" "RS_log_stream" {
+    name = "RedshiftDelivery"
+    log_group_name = "${aws_cloudwatch_log_group.log_group.name}"
+}
+
+resource "aws_cloudwatch_log_stream" "S3_log_stream" {
+    name = "S3Delivery"
+    log_group_name = "${aws_cloudwatch_log_group.log_group.name}"
+}
+
+resource "aws_kinesis_firehose_delivery_stream" "rs_stream" {
+    name = "${var.env_name}-${var.stream_name}"
+    destination = "redshift"
+
+    kinesis_source_configuration {
+      kinesis_stream_arn = "${var.datastream_source_arn}"
+      role_arn = "${aws_iam_role.firehose_to_redshift.arn}"
+    }
+
+    # this is the configuration for the intermediate s3 destination
+    s3_configuration {
+        role_arn = "${aws_iam_role.firehose_to_redshift.arn}"
+        bucket_arn = "${var.s3_intermediate_bucket_arn}"
+        buffer_size = "${var.s3_intermediate_buffer_size}"
+        buffer_interval = "${var.s3_intermediate_buffer_interval}"
+        compression_format = "GZIP"
+        kms_key_arn = "${var.s3_temp_key_arn}"
+        prefix = "${var.s3_intermediate_bucket_prefix}"
+    }
+
+    redshift_configuration {
+        role_arn = "${aws_iam_role.firehose_to_redshift.arn}"
+        cluster_jdbcurl = "${var.redshift_jdbc}"
+        username = "${var.redshift_username}"
+        password = "${var.redshift_password}"
```

Comment:
Same comment from other PR about not putting secrets in Terraform.

Reply:
See notes in identity-devops about secrets for this service.